src/Detector/Hugo/StaticSiteGeneratorHelper.cs

```diff
@@ -0,0 +1,131 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Oryx.Detector.Hugo
+{
+    /// <summary>
+    /// Helper class for functions around static site generators.
+    /// </summary>
+    internal static class StaticSiteGeneratorHelper
+    {
+        private static readonly string[] HugoConfigurationVariables =
+        {
+            "archetypeDir",
+            "baseURL",
+            "contentDir",
+            "languageCode",
+            "layoutDir",
+            "staticDir",
+            "title",
+            "theme",
+        };
+
+        /// <summary>
+        /// Checks whether or not the given repository uses a static site generator.
+        /// </summary>
+        /// <param name="sourceRepo">Source repo for the application.</param>
+        /// <returns>True if the app uses a static site generator, false otherwise.</returns>
+        public static bool IsStaticSite(ISourceRepo sourceRepo, HugoDetectorOptions options)
+        {
+            return IsHugoApp(sourceRepo, options);
+        }
+
+        /// <summary>
+        /// Checks whether or not the given repository is a Hugo application.
+        /// </summary>
+        /// <param name="sourceRepo">Source repo for the application.</param>
+        /// <param name="environment">Environment abstraction.</param>
+        /// <returns>True if the app is a Hugo app, false otherwise.</returns>
+        public static bool IsHugoApp(ISourceRepo sourceRepo, HugoDetectorOptions options)
+        {
+            // Check for Hugo environment variables
+            if (options.HasHugoEnvironmentVariables)
+            {
+                return true;
+            }
```

Comment:
imo, this HugoDetectorOptions  will be within BuildScriptGenerator instead of within detector library

Reply:
Like as part of version resolving of Hugo platform.

Reply:
Hmm, why do you think HugoDetector should not have it's own options? It is possible for a detector could have it's own settings. That said I do not like that we say that an app is a Hugo app if just this environment variable exists. We should ideally remove it.